code/script.js

```diff
@@ -1,10 +1,79 @@
 // Variables that point to selected DOM elements
 const chat = document.getElementById('chat');
+const form = document.getElementById('name-form');
+const nameInput = document.getElementById('name-input');
+const inputWrapper = document.getElementById('input-wrapper');
 
-// If you need any global variables that you can use across different functions, declare them here:
 
+// Function handling name input
+
+const handleNameInput = (event) => {
+  event.preventDefault()
+  const name = nameInput.value
+  showMessage(name, 'user')
+  nameInput.value = ''
+
+  setTimeout(() => likingMusic(name), 1000 )
+}
+
+//event listner method is starting function handleNameInput
+form.addEventListener("submit", handleNameInput);
+
+//starting question
+const likingMusic = (name) => {
+  showMessage(`Hi ${name}. Do you like listening to music?`, 'bot')
+  setTimeout(() => yesOrNo(), 100)
+}
+// two options to choose from
+
+const yesOrNo = () => {
+  inputWrapper.innerHTML = `
+  <button id="optionOne">YES</button>
+  <button id="optionTwo">NO</button>
+  `
+  document.getElementById('optionOne').addEventListener("click", () => {
+    showMessage(`Great, what is your favourite genre?`, 'bot')
+     inputWrapper.innerHTML = '' // this will remove button YES from 
+    setTimeout (() => genreQuestion('optionOne'), 100)
+  })
+
+  document.getElementById('optionTwo').addEventListener("click", () => {
+    showMessage(`Ok, it seems like there is nothing for me to do here. Good bye" 🎮 ❌`, 'user')
+    inputWrapper.innerHTML = '' // this is removing NO button
+    setTimeout (() => genreQuestio('optionTwo'), 100)
+  })
+}
+
+// Genre questions
+const genreQuestion = () => {
+  inputWrapper.innerHTML = `
+  <button id="optionPop">POP 🎤🎤🎤</button>
+  <button id="optionRock">ROCK-ROLL 🎸🎸🎸</button>
+  <button id="optionClassical">CLASSICAL 🎻🎻🎻</button>
+  `
+  document.getElementById('optionPop').addEventListener("click", () => {
+    showMessage(`Wow, you like POP music 🧑‍🎤`, 'bot')
+    showMessage(`Check out "Superstition" by Stevie Wonder 🧑`, 'bot')
+    inputWrapper.innerHTML = '' 
+    setTimeout(() => nextFuntion('optionPop'), 2000)
+  })
+
+  document.getElementById('optionRock').addEventListener("click", () => {
+    showMessage(`Great, you like Rock music 🤘`, 'bot')
+    showMessage(`Check out "Stairway to Heaven" by Led Zeppelin 🤘`, 'bot')
+    inputWrapper.innerHTML = '' 
+    setTimeout(() => nextFuntion('optionRock'), 1000)
+  })
+
+  document.getElementById('optionClassical').addEventListener("click", () => {
```

Comment:
//you could add the user answer on screen in the conversation//
showMessage('Classical', 'user')